miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BarcodableSearchRestController.java

```diff
@@ -0,0 +1,72 @@
+package uk.ac.bbsrc.tgac.miso.webapp.controller.rest;
+
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isStringEmptyOrNull;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import uk.ac.bbsrc.tgac.miso.dto.BarcodableDto;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.service.BarcodableViewService;
+
+@Controller
+@RequestMapping("/rest/barcodable")
+public class BarcodableSearchRestController extends RestController {
+  private static final String baseUrl = "/miso";
+
+  @Autowired
+  private BarcodableViewService barcodableViewService;
+
+  @RequestMapping(value = "/search", method = RequestMethod.GET)
+  @ResponseBody
+  public List<BarcodableDto> search(@RequestParam(name = "q", required = false) String query) throws IOException {
+    if (isStringEmptyOrNull(query)) {
+      return new ArrayList<>();
+    }
+
+    return barcodableViewService.search(query).stream().map(Dtos::asDto).map(this::addUrl).collect(Collectors.toList());
+  }
+
+  private BarcodableDto addUrl(BarcodableDto dto) {
+    long id = dto.getId();
+
+    switch (dto.getEntityType()) {
+    case SAMPLE:
+      dto.setUrl(makeUrl("sample", id));
+      break;
+    case BOX:
+      dto.setUrl(makeUrl("box", id));
+      break;
+    case POOL:
+      dto.setUrl(makeUrl("pool", id));
+      break;
+    case LIBRARY:
+      dto.setUrl(makeUrl("library", id));
+      break;
+    case DILUTION:
+      dto.setUrl(makeUrl("library", id));
```

Comment:
And this one looks wrong...

Reply:
Since Dilutions don't have their own page proper, we link to the associated Library.  Unintuitive, I agree.